crates/pipeline-manager/src/config.rs

```diff
@@ -217,6 +217,143 @@ pub struct CommonConfig {
+                .with_no_client_auth()
+                .with_single_cert(cert_chain, key_der)
+                .expect(
+                    "server configuration should be built using the certificate and private key",
+                );
+
+            Some(server_config)
+        } else {
+            None
+        }
+    }
+
+    /// Creates `awc` client.
+    ///
+    /// - If HTTPS is enabled for Feldera HTTP servers, the client will only have our HTTPS certificate
+    ///   as the only valid one. As such, it will be only able to connect to Feldera HTTPS servers.
+    ///   Unfortunately, it is not possible to configure `awc` to refuse to connect over HTTP at all.
+    ///
+    /// - If HTTPS is not enabled for Feldera HTTP servers, it will return the default client which can
+    ///   connect to both HTTP and HTTPS (for any valid system certificates).
+    pub fn awc_client(&self) -> awc::Client {
+        if let Some((https_tls_cert_path, _)) = self.https_config() {
+            let cert_chain: Vec<_> = CertificateDer::pem_file_iter(https_tls_cert_path)
+                .expect("HTTPS TLS certificate should be read")
+                .flatten()
+                .collect();
+            let mut root_cert_store = RootCertStore::empty();
+            for certificate in cert_chain {
+                root_cert_store.add(certificate).unwrap();
+            }
+            let config = ClientConfig::builder()
+                .with_root_certificates(root_cert_store)
+                .with_no_client_auth();
+            awc::Client::builder()
+                .connector(awc::Connector::new().rustls_0_23(Arc::new(config)))
+                .finish()
+        } else {
+            awc::Client::new()
+        }
+    }
+
+    /// Creates `reqwest` client.
+    ///
+    /// - If HTTPS is enabled for Feldera HTTP servers, the client will only have our HTTPS certificate
+    ///   as the only valid one. As such, it will be only able to connect to Feldera HTTPS servers.
+    ///   It will refuse to connect over HTTP at all.
+    ///
+    /// - If HTTPS is not enabled for Feldera HTTP servers, it will return the default client which can
+    ///   connect to both HTTP and HTTPS (for any valid system certificates).
+    pub async fn reqwest_client(&self) -> reqwest::Client {
+        if let Some((https_tls_cert_path, _)) = self.https_config() {
+            let cert_pem = tokio::fs::read_to_string(https_tls_cert_path)
+                .await
+                .expect("HTTPS TLS certificate should be read");
+            let certificate = Certificate::from_pem(cert_pem.as_bytes())
+                .expect("HTTPS TLS certificate should be readable");
+            reqwest::ClientBuilder::new()
+                .https_only(true) // Only connect to HTTPS
+                .add_root_certificate(certificate) // Add our own TLS certificate which is used
+                .tls_built_in_root_certs(false) // Other TLS certificates are not used
```

Comment:
I wonder if there is a future scenario (I can't think of one yet) where this turns out to be wrong